conn.go

```diff
@@ -278,6 +278,7 @@ func (c *Conn) Read(b []byte) (int, error) {
 func (c *Conn) ReadMessage(maxBytes int) (Message, error) {
 	batch := c.ReadBatch(1, maxBytes)
 	msg, err := batch.ReadMessage()
+	msg.Watermark = batch.Watermark()
```

Comment:
I also don't feel so good about exposing that value in the Message, it's not a property of the message. It's OK if you have to use ReadBatch to get that value, all production code should use this.